packages/ui/src/AppSidebar/AppSidebarFavoriteButton/AppSidebarFavoriteButton.tsx

```diff
@@ -0,0 +1,57 @@
+import React, { MouseEvent, useContext, useEffect, useMemo } from 'react'
+import cn from 'classnames'
+import { Star } from 'react-feather'
+
+import { IconButton } from '../../IconButton'
+import { appSidebarContext } from '../appSidebarContext'
+
+import styles from './AppSidebarFavoriteButton.module.scss'
+
+export interface AppSidebarFavoriteButtonProps {
+  id: string
+  title: string
+  className?: string
+  registrable?: boolean
+}
+
+export const AppSidebarFavoriteButton = ({ id, title, className, registrable }: AppSidebarFavoriteButtonProps) => {
+  const { favorites, setFavorites, registerFavorite, unregisterFavorite } = useContext(appSidebarContext)
+  const isFavoriteItem = useMemo(() => favorites.find((item) => item === id), [favorites, id])
+
+  const handleToggleFavorite = (e: MouseEvent) => {
+    e.preventDefault()
```

Comment:
Why do we need `preventDefault()` here?

Reply:
removed

Reply:
But i think we need to stop propagation to prevent navigating to the page when we click the star icon

Reply:
Ok, I checked `AppSidebar.stories.tsx` and I navigated to `/` after clicking `Test` item. It's wrapped with an `<a>`, so this should be fixed somehow, but not within this event handler.
```js
<a href="/">
        <AppSidebarItem id="1" title="Test" active />
 </a>
```

Reply:
> Ok, I checked `AppSidebar.stories.tsx` and I navigated to `/` after clicking `Test` item. It's wrapped with an `<a>`, so this should be fixed somehow, but not within this event handler.
> 
> ```js
> <a href="/">
>         <AppSidebarItem id="1" title="Test" active />
>  </a>
> ```

`<AppSidebarItem id="1" title="Test" active wrapper={({children}) => <a href="/">{children}</a>} />`